woocommerce/woocommerce
# WooCommerce Docs: Improve Manifest Structure, Extend Frontmatter Support

### Changes proposed in this Pull Request:

This PR introduces a few changes:

1. I've improved the manifest generator category handling. If it finds a directory with no `README.md` it still creates a category and gives it a title that is the capitalized folder name. Before the manifest generator could not handle a missing README.md

2. I've changed the frontmatter generation in the manifest generator to better reflect where frontmatter is changing post attributes (e.g. `title` is now `post_title`). We only allow a few attributes right now. I've also changed the category frontmatter to indicate that it's category specific (`title` is now `category_title`).

3. I've added support for documents to have an edit_link property, if you're using github this will directly to an edit link in Github, right now if you're not using github this link won't be generated. Later I'll implement post meta to store this info on posts in the plugin.

4. In the WooCommerce docs plugin I've updated its parsing for the new format and frontmatter.

5. More supporting tests




### How to test the changes in this Pull Request:

Testing the Plugin:

1. Have the docs plugin mounted in your dev environment.
2. Build the docs plugin, via pnpm start
3. Now visit the menu item for the plugin in wp-admin (called WooCommerce Docs)
4. Add a new manifest URL via the basic UI, it will need to point to the sample manifest that is included with the plugin. The URL should be like (if in docker) http://host.docker.internal:PORT_NUMBER/wp-content/plugins/woocommerce-docs/scripts/manifest.json
5. The action scheduler job runs once a minute after activation of the plugin. Once it has run it should have generated a set of categories and posts with content matching that of the sample docs in example-docs folder that is bundled with the plugin currently.
6. Run the php unit tests (still not hooked up in CI yet but I will do this soon as a separate PR): `pnpm test:unit`

Testing the CLI:

1. Navigate to the `tools/monorepo-utils` directory and ensure that the CLI is built: `pnpm build`
2. Now from Monorepo root run `pnpm utils md-docs create ./plugins/woocommerce-docs/example-docs woodocs --outputFilePath ./plugins/woocommerce-docs/scripts/manifest.json` A manifest should be generated without error.
3. Run the unit tests locally (still not hooked up in CI yet, but I will do this soon as separate PR) `pnpm test`


### Changelog entry




<details>

#### Significance




#### Type




#### Message 

#### Comment 

</details>
